packages/rn-tester/js/examples/TVFocusGuide/TVFocusGuideAutoFocusExample.js

```diff
@@ -0,0 +1,323 @@
+      <FocusableBox style={styles.sideMenuItem} />
+      <FocusableBox style={styles.sideMenuItem} />
+      <FocusableBox style={styles.sideMenuItem} />
+    </TVFocusGuide>
+  );
+};
+
+const getItemText = ({item, prefix}) => {
+  return prefix ? `${prefix}-${item}` : `${item}`;
+};
+
+const HList = ({
+  itemCount,
+  itemWidth = 500 * scale,
+  itemHeight = 120 * scale,
+  itemColor = FOCUSABLE_ITEM_COLOR,
+  onItemFocused,
+  onItemPressed,
+  prefix = '',
+  ...props
+}) => {
+  const listRef = React.useRef(null);
+
+  const data = React.useMemo(() => generateData(itemCount), [itemCount]);
+
+  const renderItem = ({item, index}) => {
+    return (
+      <FocusableBox
+        width={itemWidth}
+        height={itemHeight}
+        color={itemColor}
+        style={styles.mr5}
+        text={getItemText({prefix, item})}
+        onFocus={() => onItemFocused?.({item, index})}
+        onPress={() => onItemPressed?.({item, index})}
+      />
+    );
+  };
+
+  return (
+    <FlatList
+      keyExtractor={item => getItemText({prefix, item})}
+      ref={listRef}
+      data={data}
+      renderItem={renderItem}
+      horizontal
+      contentContainerStyle={styles.hListContainer}
+      {...props}
+    />
+  );
+};
+
+const categoryData = [1, 2, 3, 4, 5];
+const getSelectedItemPrefix = selectedCategory => {
+  if (selectedCategory === null) return 'Item';
+  return `Category ${selectedCategory} - Item`;
+};
+const Row = ({title}) => {
+  const [selectedCategory, setSelectedCategory] = React.useState('1');
+
+  const onCategoryFocused = ({item}) => {
+    setSelectedCategory(item);
+  };
+
+  return (
+    <View style={styles.mb5}>
+      <TVFocusGuide autoFocus style={styles.rowTop}>
+        <Text style={styles.rowTitle}>{title}</Text>
+        <HList
+          prefix="Category"
+          itemCount={5}
+          data={categoryData}
+          itemHeight={50 * scale}
+          itemWidth={200 * scale}
+          onItemFocused={onCategoryFocused}
+        />
+      </TVFocusGuide>
+      <TVFocusGuide autoFocus style={styles.mb5}>
+        <HList
+          prefix={getSelectedItemPrefix(selectedCategory)}
+          itemCount={10}
+        />
+      </TVFocusGuide>
+    </View>
+  );
+};
+
+const Col = ({title}) => {
+  return (
+    <TVFocusGuide autoFocus style={styles.col}>
+      <Text style={styles.rowTitle}>{title}</Text>
+      <FocusableBox text="0" style={styles.colItem} />
+      <FocusableBox text="1" style={styles.colItem} />
+      <FocusableBox text="2" style={styles.colItem} />
+      <FocusableBox text="3" style={styles.colItem} />
+    </TVFocusGuide>
+  );
+};
+
+const RestoreFocusTestList = () => {
+  const [randomize, setRandomize] = React.useState(false);
+  const data = React.useMemo(() => generateData(10, randomize), [randomize]);
+  /**
+   * This is a test to make sure that the focus is restored
+   * after the list is re-rendered and currently focused item is removed.
```

Comment:
I tested this and it works quite well on both platforms.

I did notice that when the list is rerendered, the focus goes to the leftmost item currently on screen, not the first item in the list -- this is probably what we want anyway as it preserves horizontal scroll position.

Reply:
Yeah, this is known and expected. The scroll state should be reset first if you want it to move the focus to the *very* first item of the list. Scrollable content (especially virtualized) can lead to interesting behavior as I pointed out in the description. It's a separate problem that I'll take a look at later, though. Just wanted to leave a comment here for future readers.